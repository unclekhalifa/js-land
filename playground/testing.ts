console.log('confirming this works')