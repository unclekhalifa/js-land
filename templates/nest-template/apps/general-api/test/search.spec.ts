import { Test } from '@nestjs/testing'
import { AppModule } from '../src/app/app.module'
import { INestApplication } from '@nestjs/common'
import request from 'supertest'

describe('Search', () => {
  let app: INestApplication
  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
  let server: any

  async function setup() {
    const module = await Test.createTestingModule({
      imports: [AppModule],
      providers: []
    }).compile()

    app = module.createNestApplication()
    app.setGlobalPrefix('api')

    await app.init()
    server = app.getHttpServer()
  }

  beforeAll(async () => {
    await setup()
  })

  afterAll(async () => {
    await app.close()
  })

  describe('GET /search/companies', () => {
    it('should return a company given a number', async () => {
      const res = await request(server).get('/api/search/companies?number=12345678')
      expect(res.status).toBe(200)
      expect(res.body).toMatchSnapshot()
    })

    it('should not return a company if given wrong params', async () => {
      const res = await request(server).get('/api/search/companies?random=some-string')
      expect(res.status).toBe(400)
    })

    it('should return 400 if missing query params', async () => {
      const res = await request(server).get('/api/search/companies')
      expect(res.status).toBe(400)
    })
  })

  describe('GET /search/companies/:companyNumber', () => {
    it('should return a company given a company number', async () => {
      const res = await request(server).get('/api/search/companies/12345678')

      expect(res.status).toBe(200)
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('GET /search/officers', () => {
    it('should return an officer given a number', async () => {
      const res = await request(server).get('/api/search/officers?number=12345678')
      expect(res.status).toBe(200)
      expect(res.body).toMatchSnapshot()
    })

    it('should not return an officer if given wrong params', async () => {
      const res = await request(server).get('/api/search/officers?random=some-string')
      expect(res.status).toBe(400)
    })

    it('should return 400 if missing query params', async () => {
      const res = await request(server).get('/api/search/officers')
      expect(res.status).toBe(400)
    })
  })

  describe('GET /search/officers/:officerNumber', () => {
    it('should return an officer given an officer number', async () => {
      const res = await request(server).get('/api/search/officers/12345678')

      expect(res.status).toBe(200)
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('GET /search/ai', () => {
    it('should return a response', async () => {
      const res = await request(server).get('/api/search/ai?text=some-string')

      expect(res.status).toBe(200)
      expect(res.body).toMatchSnapshot()
    })
  })

})
