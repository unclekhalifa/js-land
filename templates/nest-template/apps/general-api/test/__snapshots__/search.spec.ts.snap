// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search GET /search/ai should return a response 1`] = `
{
  "response": "some-string",
}
`;

exports[`Search GET /search/companies should return a company given a number 1`] = `
[
  {
    "name": "some name",
    "number": "12345678",
  },
]
`;

exports[`Search GET /search/companies/:companyNumber should return a company given a company number 1`] = `
{
  "name": "some-name",
  "number": "12345678",
}
`;

exports[`Search GET /search/officers should return an officer given a number 1`] = `
[
  {
    "name": "some name",
    "number": "12345678",
  },
]
`;

exports[`Search GET /search/officers/:officerNumber should return an officer given an officer number 1`] = `
{
  "name": "some-name",
  "number": "12345678",
}
`;
